File: src/shared/constants/bots.ts
export const BOT_IDS = ["c_level_executive", "data_analyst"] as const;

export type BotId = (typeof BOT_IDS)[number];

export type BotDefinition = {
  id: BotId;
  label: string;
  description: string;
  icon: "briefcase" | "chart";
  inputPlaceholder: string;
};

export const DEFAULT_BOT_ID: BotId = "c_level_executive";

export const BOT_DEFINITIONS: Record<BotId, BotDefinition> = {
  c_level_executive: {
    id: "c_level_executive",
    label: "C‑Level Executive",
    description: "Brief, decision-oriented insights with risks and next actions.",
    icon: "briefcase",
    inputPlaceholder: "Ask for an executive summary...",
  },
  data_analyst: {
    id: "data_analyst",
    label: "Data Analyst",
    description: "More technical, methodical analysis with assumptions and follow-ups.",
    icon: "chart",
    inputPlaceholder: "Ask for analysis (metrics, breakdowns, methodology)...",
  },
};

File: backend/bots.py
"""Bot persona definitions for analysis explanations."""

from __future__ import annotations

from dataclasses import dataclass
from typing import Literal

BotId = Literal["c_level_executive", "data_analyst"]


@dataclass(frozen=True)
class BotDefinition:
    id: BotId
    label: str
    explain_prompt_template: str


BOT_DEFINITIONS: dict[BotId, BotDefinition] = {
    "c_level_executive": BotDefinition(
        id="c_level_executive",
        label="C-Level Executive",
        explain_prompt_template="""
You are a C-Level Executive Advisor.

Context (industry): {industry}  
User question: {query}  
Computed result: {result}

Return a **clean, professional Markdown response** using the structure below.
Follow these formatting rules strictly:
- Use `##` for section headers
- Use concise bullets (`-`)
- Leave one blank line between sections
- Keep language executive-level and decisive
- Avoid filler or repetition

---

## Executive Summary
(2–4 sentences. High-level, outcome-focused.)

## Key Findings
- Bullet points only
- Include numbers, percentages, or magnitudes where possible
- Focus on material insights

## Risks / Caveats
- Key uncertainties, assumptions, or downside risks
- Be realistic and succinct

## Recommended Actions
1. Most important action first
2. Action-oriented and specific
3. Feasible within the stated context

---

### Strategic Insight
(One sharp, forward-looking sentence that reframes the situation.)
""".strip(),
    ),
    "data_analyst": BotDefinition(
        id="data_analyst",
        label="Data Analyst",
        explain_prompt_template="""
You are a Data Analyst.

Context (industry): {industry}  
User question: {query}  
Computed result: {result}

Return a **precise, well-structured Markdown response**.
Formatting rules:
- Use `##` for section headers
- Use bullets for lists
- Use short, information-dense sentences
- No executive language or buzzwords
- Clarity and methodological rigor are mandatory

---

## Approach
- How the question was interpreted
- What data, metrics, or logic were applied
- Any transformations or calculations performed

## Findings
- Key results in bullet form
- Include numeric values, ranges, distributions, or segments where applicable
- State what was observed, not what it “means”

## Assumptions & Data Quality Notes
- Explicit assumptions
- Known limitations, gaps, or reliability concerns

## Suggested Next Analyses
- 3–5 concrete follow-up analyses
- Each should clearly extend or validate the findings
""".strip(),
    ),
}


DEFAULT_BOT_ID: BotId = "c_level_executive"


def normalize_bot_id(value: object) -> BotId:
    """Return a supported bot id; fall back to default for unknown values."""
    if value == "c_level_executive":
        return "c_level_executive"
    if value == "data_analyst":
        return "data_analyst"
    return DEFAULT_BOT_ID
